.github/workflows/publish.yml

```diff
@@ -0,0 +1,37 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+# Controls when the workflow will run
+on:
+  push:
+    branches:
+      - master
+
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: Publish package
+        run: |
+          mkdir -p ~/.gradle
+          ./gradlew -i publishToMavenCentral
```

Comment:
This step was failing for me when I was testing it locally (with `nektos/act`) with a gradle problem (as opposed to an expected `401`):

```
| Error: Could not find or load main class org.gradle.wrapper.GradleWrapperMain
| Caused by: java.lang.ClassNotFoundException: org.gradle.wrapper.GradleWrapperMain
```

The error seems to indicate that the CI isn't able to find the `gradle-wrapper.jar`, which is odd considering it's definitely been added to git.

Looking at .gitignore, we see that `*.jar` is ignored. This shouldn't have an effect on already-added `.jar` files (i.e. `gradle-wrapper.jar`).

Despite this, when I removed `*.jar` from `.gitignore`, I was able to get past this stage:

```
Failed to load staging profiles, server at https://oss.sonatype.org/service/local/ responded with status code 401, body:
```

I don't see a good reason to keep `*.jar` in our `.gitignore` - @al-jeyapal  what do you think?

Reply:
Comment hasn't been addressed yet, can't close this out